src/main/java/io/neonbee/internal/processor/EntityCollectionProcessor.java

```diff
@@ -86,87 +83,91 @@ public void readEntityCollection(ODataRequest request, ODataResponse response, U
          */
         EdmEntitySet edmEntitySet = ((UriResourceEntitySet) uriInfo.getUriResourceParts().get(0)).getEntitySet();
         EdmEntityType edmEntityType = edmEntitySet.getEntityType();
+
         // Fetch the data from backend
-        Promise<Void> processPromise = getProcessPromise();
+        fetchEntities(request, edmEntityType, ew -> {
+            try {
+                List<Entity> resultEntityList = applyFilterQueryOption(uriInfo.getFilterOption(), ew.getEntities());
+
+                if (!resultEntityList.isEmpty()) {
+                    applyOrderByQueryOption(uriInfo.getOrderByOption(), resultEntityList);
+                    resultEntityList = applySkipQueryOption(uriInfo.getSkipOption(), resultEntityList);
+                    resultEntityList = applyTopQueryOption(uriInfo.getTopOption(), resultEntityList);
+                    entityCollection.getEntities().addAll(resultEntityList);
+                }
+
+                EntityCollectionSerializerOptions opts = createSerializerOptions(request, uriInfo, edmEntitySet);
+                response.setContent(odata.createSerializer(responseFormat)
+                        .entityCollection(serviceMetadata, edmEntityType, entityCollection, opts).getContent());
+                response.setStatusCode(HttpStatusCode.OK.getStatusCode());
+                response.setHeader(HttpHeader.CONTENT_TYPE, responseFormat.toContentTypeString());
+                getProcessPromise().complete();
+            } catch (ODataException e) {
+                getProcessPromise().fail(e);
```

Comment:
Same here

Reply:
I think it is only called once, because if it passes the try block, the catch block isn't evaluated.